apps/wallet/src/features/Send/InputTransaction/TransactionFee/index.tsx

```diff
@@ -13,13 +14,20 @@ interface Props {
 
 export const TransactionFee: FC<Props> = ({ network }) => {
 	const { tokens } = useTokens(network);
+
 	const isAbstractFee = gasilonSupportedNetworks.includes(network);
+	if (isAbstractFee) {
+		const tokenList = tokens;
+		const solIndex = tokenList.findIndex(
+			(token) => token.account.mint === solMint,
+		);
+		if (solIndex !== -1) {
+			[tokenList[0], tokenList[solIndex]] = [tokenList[solIndex], tokenList[0]];
+		}
+		return <AbstractedTransactionFee tokenList={tokens} />;
+	}
```

Comment:
First, I think a conditional return like this is not good (does not clean).
`tokenList` should be renamed to `sortedTokens` to make it clear.
Also, I am sure that `SOL` token is always placed at the head of `tokens`. Clone a state array and mutate it to passing as a prop for child component might cause side effect

Reply:
`SOL` is not placed at the head in my case (while testing, `USDC-Dev` is the first one).  I don't want to break other code so I sort it out here.

`sortedTokens` is obviously a better name, gonna change it now

Reply:
If `SOL` is not placed at the head, it might be a bug, but it was resolved on the newest engine updating